apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+                    {"p", SpannerDbType.String, s_keys[s_rand.Value.Next(s_keys.Count)]}
+                });
+                using (var reader =
+                    await cmd.ExecuteReaderAsync(timestampbound))
+                {
+                    while (await reader.ReadAsync())
+                    {
+                    }
+                }
+                latencies.Add(sw.ElapsedMilliseconds);
+            }),
+            new Operation("updateproportion", async latencies =>
+            {
+                var sw = Stopwatch.StartNew();
+                var cmd = s_connection.CreateUpdateCommand(s_table,
+                    new SpannerParameterCollection
+                    {
+                        {"id", SpannerDbType.String, s_keys[s_rand.Value.Next(s_keys.Count)]},
+                        {s_fieldNames[s_rand.Value.Next(10)], SpannerDbType.String, GenerateRandomString()}
+                    });
+                await cmd.ExecuteNonQueryAsync();
+                latencies.Add(sw.ElapsedMilliseconds);
+            }),
+            new Operation("scanproportion", x => throw new NotImplementedException("scan is not yet implemented")),
+            new Operation("insertproportion", x => throw new NotImplementedException("scan is not yet implemented"))
+        };
+
+        public static int Main(string[] args)
+        {
+            if (!TryParseArguments(args))
+            {
+                return -1;
+            }
+            LoadConfigFile();
+            ValidateArguments();
+            LoadKeys();
+            InitializeSettings();
+
+            DebugMessage("prewarm...");
+            Task.WhenAll(Enumerable.Range(1, GetOptionWithDefault(s_prewarm, 0)).Select(x => PreWarm()))
+                .WaitWithUnwrappedExceptions();
+
+            DebugMessage("running...");
+            var overallTime = Stopwatch.StartNew();
+            if (s_options.ContainsKey(s_targetQps))
+            {
+                //target qps version
+                RunWorkerWithQps(GetOption<double>(s_targetQps), TimeSpan.FromSeconds(GetOption<int>(s_qpstestTime)))
+                    .WaitWithUnwrappedExceptions();
+            }
+            else
+            {
+                //target worker version
+                RunWithTargetWorkers(GetOption<int>(s_num_worker)).WaitWithUnwrappedExceptions();
+            }
+            PrintResults(overallTime.ElapsedMilliseconds);
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Runs a single worker task
+        /// </summary>
+        private static async Task PreWarm()
+        {
+            using (var spannerConnection = new SpannerConnection(
+                $"Data Source=projects/{s_project}/instances/{s_instance}/databases/{s_database}"))
+            {
+                await spannerConnection.OpenAsync();
+            }
+        }
+
+        /// <summary>
+        /// Runs a single worker task that targets a specific QPS without regard to workers.
```

Comment:
This looks like it starts multiple worker tasks, not a single one. Or I may just be confused.

Reply:
the wording is poor.  it meant this is to be used as a single task that executes multiple subtasks itself.